### ISSUE TITLE
crash when morphing > 3 presets

### ISSUE THREAD

**Issue by hollyhook** [still-scene/t3#30]
Select blob operator
in preset window, select 4 or more presets 
hold Alt and move the mouse
  -> the presets in use seem to jump around
move mouse outside the preset window
  -> crash

While the use of morphing between that many presets is questionable, it still shouldn't crash

```
System.ArgumentOutOfRangeException
  HResult=0x80131502
  Message=Index was out of range. Must be non-negative and less than the size of the collection. Arg_ParamName_Name
  Source=System.Private.CoreLib
  StackTrace:
   at System.ThrowHelper.ThrowArgumentOutOfRange_IndexException()
   at System.Collections.Generic.List`1.get_Item(Int32 index)
   at T3.Gui.Windows.Variations.VariationBaseCanvas.TryGetBlendWeight(Variation v, Single& weight) in C:\dev\repos\t3\t3\Gui\Windows\Variations\VariationBaseCanvas.cs:line 312
   at T3.Gui.Windows.Variations.VariationThumbnail.Draw(VariationBaseCanvas canvas, Variation variation, ImDrawListPtr drawList, ShaderResourceView canvasSrv, ImRect uvRect) in C:\dev\repos\t3\t3\Gui\Windows\Variations\VariationThumbnail.cs:line 104
   at T3.Gui.Windows.Variations.VariationBaseCanvas.Draw(ImDrawListPtr drawList) in C:\dev\repos\t3\t3\Gui\Windows\Variations\VariationBaseCanvas.cs:line 67
   at T3.Gui.Windows.Variations.VariationsWindow.DrawWindowContent() in C:\dev\repos\t3\t3\Gui\Windows\Variations\VariationsWindow.cs:line 106
   at T3.Gui.Windows.Variations.VariationsWindow.DrawContent() in C:\dev\repos\t3\t3\Gui\Windows\Variations\VariationsWindow.cs:line 22
   at T3.Gui.Windows.Window.DrawOneInstance() in C:\dev\repos\t3\t3\Gui\Windows\Window.cs:line 93
   at T3.Gui.Windows.Window.Draw() in C:\dev\repos\t3\t3\Gui\Windows\Window.cs:line 53
   at T3.Gui.Windows.WindowManager.Draw() in C:\dev\repos\t3\t3\Gui\Windows\WindowManager.cs:line 94
   at T3.Gui.T3Ui.Draw() in C:\dev\repos\t3\t3\Gui\T3UI.cs:line 92
   at T3.Program.<>c__DisplayClass9_0.<Main>g__RenderCallback|1() in C:\dev\repos\t3\t3\Program.cs:line 182
   at SharpDX.Windows.RenderLoop.Run(Control form, RenderCallback renderCallback, Boolean useApplicationDoEvents)
   at T3.Program.Main() in C:\dev\repos\t3\t3\Program.cs:line 198

```

**Comment by hollyhook** [comment]
credits to sense to finding the issue in the first place

**Comment by pixtur** [comment]
Excellent report. That should be easy to fix!

**Comment by pixtur** [comment]
Fixed in v3.3.0
